core/plugins/org.polarsys.capella.core.data.core.properties/src/org/polarsys/capella/core/data/core/properties/fields/TitleBlockBasicElementGroup.java

```diff
@@ -0,0 +1,179 @@
+      contentTextField = createTextField(textGroup, Messages.getString("NamedElement.ContentLabel"));
+    }
+  }
+
+  /**
+   * @param textGroup
+   * @param textLabel
+   */
+  private Text createTextField(Group textGroup, String textLabel) {
+    widgetFactory.createCLabel(textGroup, textLabel);
+
+    Text textField = widgetFactory.createText(textGroup, ICommonConstants.EMPTY_STRING);
+    textField.addFocusListener(this);
+    textField.addKeyListener(this);
+    textField.addModifyListener(this);
+    textField.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+    return textField;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public void loadData(EObject semanticElement, String name, String content) {
+    super.loadData(semanticElement, null);
+
+    if (null != semanticElement) {
+      if (null != nameTextField)
+        setTextValue(nameTextField, name);
+      if (null != contentTextField)
+        setTextValue(contentTextField, content);
+    }
+  }
+
+  protected void setTextValue(Text text, String value) {
+    text.setText(value);
+  }
+
+  /**
+   * @param textField
+   *          text field to be filled
+   */
+  @Override
+  protected void fillTextField(Text textField) {
+    if (textField.equals(nameTextField)) {
+      setFieldValue(semanticElement, NAME, nameTextField.getText());
+    } else if (textField.equals(contentTextField)) {
+      setFieldValue(semanticElement, CONTENT, contentTextField.getText());
+    }
+  }
+
+  private void setFieldValue(EObject object, String field, final Object value) {
+    if (!((DAnnotation) object).getDetails().get(field).equals(value.toString())) {
+      DDiagram diagram = (DDiagram) object.eContainer();
+      AbstractReadWriteCommand command = new AbstractReadWriteCommand() {
+        @Override
+        public void run() {
+          ((DAnnotation) object).getDetails().put(field, value.toString());
+          RefreshRepresentationsCommand refreshCommand = new RefreshRepresentationsCommand(
+              TransactionUtil.getEditingDomain(diagram), new NullProgressMonitor(), diagram);
+          refreshCommand.execute();
+        }
+      };
+      executeCommand(command);
+    }
+  }
+
+  /**
+   * @param enabled
+   *          whether or not the name text field is enabled
+   */
+  public void enableNameField(boolean enabled) {
+    if (null != nameTextField && !nameTextField.isDisposed()) {
+      nameTextField.setEnabled(enabled);
+    }
+  }
+
+  /**
+   * @param enabled
+   *          whether or not the content text field is enabled
+   */
+  public void enableContentField(boolean enabled) {
+    if (null != contentTextField && !contentTextField.isDisposed()) {
+      contentTextField.setEnabled(enabled);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void setEnabled(boolean enabled) {
+    LockHelper.getInstance().enable(nameTextField, enabled);
+    LockHelper.getInstance().enable(contentTextField, enabled);
+  }
+
+  @Override

```

Comment:
I think this can be removed.

Reply:
Is abstract in AbstractSemanticField and thus needs to be implemented.